### PR TITLE
Add Python 2.7 support

### PR DESCRIPTION
**Pull request by sYnfo**

### CODE REVIEW

**Comment by mnagy** [comment]
[test]

**Comment by mnagy** [comment]
@mfojtik cc

File: 2.7/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
+FROM openshift/base-centos7
+
+# This image provides a Python 2.7 environment you can use to run your Python
+# applications.
+
+MAINTAINER Maciej Szulik <maszulik@redhat.com>
```
Comment (mfojtik) [inline 2.7/Dockerfile]: @sYnfo 

```
MAINTAINER SoftwareCollections.org <sclorg@redhat.com>
```

(also please apply to 3.3 dockerfiles and RHEL)

File: 2.7/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+EXPOSE 8080
+
+ENV PYTHON_VERSION=2.7 \
+    PATH=.local/bin/:$PATH
+
+LABEL k8s.io/description="Platform for building and running Python 2.7 applications" \
+      k8s.io/display-name="Python 2.7" \
+      openshift.io/expose-services="8080:http" \
+      openshift.io/tags="builder,python,python27"
```
Comment (mfojtik) [inline 2.7/Dockerfile]: add rh-python27

File: 2.7/Dockerfile.rhel7 [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+EXPOSE 8080
+
+ENV PYTHON_VERSION=2.7 \
+    PATH=.local/bin/:$PATH
+
+LABEL k8s.io/description="Platform for building and running Python 2.7 applications" \
+      k8s.io/display-name="Python 2.7" \
+      openshift.io/expose-services="8080:http" \
+      openshift.io/tags="builder,python,python27"
```
Comment (mfojtik) [inline 2.7/Dockerfile.rhel7]: add rh-python27

**Comment by mfojtik** [comment]
@sYnfo please note that you will have to rebase this when https://github.com/openshift/sti-python/pull/43 lands (mostly fixes the syntax of labels)

File: 2.7/.sti/bin/usage [hunk 4]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+DISTRO=`cat /etc/*-release | grep ^ID= | grep -Po '".*?"' | tr -d '"'`
+
+cat <<EOF
+This is a STI python-2.7 ${DISTRO} base image:
+To use it, install STI: https://github.com/openshift/source-to-image
```
Comment (mfojtik) [inline 2.7/.sti/bin/usage]: Please change STI to S2I while you're here :-)

**Comment by sYnfo** [comment]
Rebased, fixed.

**Comment by mfojtik** [comment]
@rhcarvalho || @soltysh do you guys want to have second look? (LGTM to me now)

**Comment by rhcarvalho** [comment]
In v2 we avoid duplicating scripts by having [common code used by 2.7 and 3.3](https://github.com/openshift/origin-server/tree/master/cartridges/openshift-origin-cartridge-python/usr/versions). I miss something like that here.

Copy-pasting all scripts `assemble`, `run`, and `usage` and the tests seems unnecessary and will make it more complicated to maintain.

**Comment by mfojtik** [comment]
@rhcarvalho this is more general problem we will have to solve for all images. I think for now it is OK to keep the copies, but in future we would most likely be doing symlinks from 'common' folder. It's not a blocker for this PR, but something we have to solve globally.

**Comment by rhcarvalho** [comment]
I think it's easier to don't introduce duplication than to do it and then have to figure out the (sometimes unintended) differences 2.7 and 3.3 supporting scripts might have in the future when unifying the common parts.

**Comment by rhcarvalho** [comment]
Comparing `2.7/Dockerfile` to `3.3/Dockerfile` I see that in 2.7 the label `io.openshift.tags` includes `rh-python27` while 3.3 has nothing equivalent

File: 2.7/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+LABEL io.k8s.description="Platform for building and running Python 2.7 applications" \
+      io.k8s.display-name="Python 2.7" \
+      io.openshift.expose-services="8080:http" \
+      io.openshift.tags="builder,python,python27,rh-python27"
+
+RUN yum install -y \
+    https://www.softwarecollections.org/en/scls/rhscl/python27/epel-7-x86_64/download/rhscl-python27-epel-7-x86_64.noarch.rpm && \
+    yum install -y --setopt=tsflags=nodocs --enablerepo=centosplus python27 python27-python-devel python27-python-setuptools python27-python-pip epel-release && \
+    yum install -y --setopt=tsflags=nodocs install nss_wrapper && \
+    yum clean all -y && \
```
Comment (rhcarvalho) [inline 2.7/Dockerfile]: Delete ` && \`?

**Comment by mfojtik** [comment]
@rhcarvalho we decided to add that tag to sync with 'rhcsl' product name (that changed in new version of rhcsl). No need to add that tag into older image.

**Comment by bparees** [comment]
just needs @rhcarvalho 's comment about the && \ addressed and commits squashed and then lgtm.

**Comment by sYnfo** [comment]
Fixed trailing characters and squashed, lemme know if it looks good now. :)

**Comment by rhcarvalho** [comment]
[test]

**Pull request merged**